src/importexport/musicxml/internal/musicxml/export/exportmusicxml.cpp

```diff
@@ -7618,26 +7622,39 @@ static void partList(XmlWriter& xml, Score* score, MusicXmlInstrumentMap& instrM
         xml.startElementRaw(String(u"score-part id=\"P%1\"").arg(idx + 1));
         initInstrMap(instrMap, part->instruments(), score);
         static const std::wregex acc(L"[♭♯]");
-        XmlWriter::Attributes attributes;
+        XmlWriter::Attributes longInstrumentAttributes;
+        XmlWriter::Attributes shortInstrumentAttributes;
         // by default export the parts long name as part-name
         String partName = part->longName();
         // use the track name if no part long name
         if (partName.empty()) {
             partName = part->partName();
             if (!partName.empty()) {
-                attributes.emplace_back(std::make_pair("print-object", "no"));
+                hiddenInstrName = true;
             }
         }
-        xml.tag("part-name", attributes, MScoreTextToMusicXml::toPlainText(partName).replace(u"♭", u"b").replace(u"♯", u"#"));
+        if (hiddenInstrName) {
+            longInstrumentAttributes.push_back({ "print-object", "no" });
+            shortInstrumentAttributes.push_back({ "print-object", "no" });
+        }
+        if (longInstrumentColor != engravingConfiguration()->defaultColor()) {
+            longInstrumentAttributes.push_back({ "color", String::fromStdString(longInstrumentColor.toString()) });
+        }
+        xml.tag("part-name", longInstrumentAttributes,
+                MScoreTextToMusicXml::toPlainText(partName).replace(u"♭", u"b").replace(u"♯", u"#"));
```

Comment:
Change to
```cpp
                MScoreTextToMusicXml::toPlainText(partName).replace(u'♭', 'b').replace(u'♯', '#'));
``
`to replace single characters rather than strings (of length 1) )single quotes, and `u` only for the unicode chars)